keras/src/wrappers/_sklearn.py

```diff
@@ -0,0 +1,485 @@
+                The targets.
+            **kwargs: keyword arguments passed to `model.fit`
+        """
+        X, y = _validate_data(self, X, y)
+        y = self._process_target(y, reset=True)
+        model = self._get_model(X, y)
+        _check_model(model)
+
+        rand = self._get_random_int()
+        fit_args = self.fit_args or {}
+        fit_args.update(kwargs)
+        if rand is not None:
+            with tensorflow_random_state(rand):
+                self.history_ = model.fit(X, y, **fit_args)
+        else:
+            self.history_ = model.fit(X, y, **fit_args)
+
+        self.model_ = model
+        return self
+
+    def predict(self, X):
+        """Predict using the model."""
+        check_is_fitted(self)
+        X = _validate_data(self, X, reset=False)
+        raw_output = self.model_.predict(X)
+        return self._reverse_process_target(raw_output)
+
+    def _process_target(self, y, reset=False):
+        """Regressors are NOOP here, classifiers do OHE."""
+        # This is here to raise the right error in case of invalid target
+        type_of_target(y, raise_unknown=True)
+        if reset:
+            self._target_encoder = TargetReshaper().fit(y)
+        return self._target_encoder.transform(y)
+
+    def _reverse_process_target(self, y):
+        """Regressors are NOOP here, classifiers reverse OHE."""
+        return self._target_encoder.inverse_transform(y)
+
+
+@keras_export("keras.wrappers.SKLearnClassifier")
+class SKLearnClassifier(ClassifierMixin, SKLBase):
+    """scikit-learn compatible classifier wrapper for Keras models.
+
+    Args:
+        model: `Model`.
+            An instance of `Model`, or a callable returning such an object.
+            Note that if input is a `Model`, it will be cloned using
+            `keras.models.clone_model` before being fitted, unless
+            `warm_start=True`.
+            The `Model` instance needs to be passed as already compiled.
+            If callable, it must accept at least `X` and `y` as keyword
+            arguments. Other arguments must be accepted if passed as
+            `model_args` by the user.
+        warm_start: bool, defaults to False.
+            Whether to reuse the model weights from the previous fit. If `True`,
+            the given model won't be cloned and the weights from the previous
+            fit will be reused.
+        model_args: dict, defaults to None.
+            Keyword arguments passed to `model`, if `model` is callable.
+        fit_args: dict, defaults to None.
+            Keyword arguments passed to `model.fit`. These can also be passed
+            directly to the `fit` method of the scikit-learn wrapper. The
+            values passed directly to the `fit` method take precedence over
+            these.
+        random_state : int, np.random.RandomState, or None, defaults to None.
+            Set the Tensorflow random number generators to a reproducible
```

Comment:
This should work with all backends, not just TF. So no mention of TF and no special casing of TF in the code